kernel/aster-nix/src/net/socket/ip/stream/mod.rs

```diff
@@ -589,6 +591,63 @@ impl Socket for StreamSocket {
         });
         Ok(())
     }
+
+    fn sendmsg(&self, msg_hdr: MessageHeader, flags: SendRecvFlags) -> Result<usize> {
+        let MessageHeader {
+            addr,
+            io_vecs,
+            control_message,
+        } = msg_hdr;
+
+        if control_message.is_some() {
+            // TODO: support sending control message
+            warn!("sending control message is not supported");
+        }
+
+        let mut total_bytes = 0;
+        for io_vec in IoVecIter::new(&io_vecs) {
+            if io_vec.is_empty() {
+                continue;
+            }
+
+            let buffer = {
+                let mut buffer = vec![0u8; io_vec.len()];
+                io_vec.read_from_user(&mut buffer)?;
+                buffer
+            };
+
+            // Remote address is ignored
+            let sent_bytes = self.sendto(&buffer, None, flags)?;
```

Comment:
There should be issues that are similar to `recvmsg`, please also check and fix them.

Reply:
Not sure whether `sendmsg`/`recvmsg` should be atomic. If it needs to be atomic, we cannot invoke `try_recvfrom`/`try_sendto` multiple times for both TCP streams and UDP datagrams, unfortunately.

Note that `readv`/`writev` must be atomic, according to their [man pages](https://man7.org/linux/man-pages/man2/readv.2.html), as quoted below. But I didn't find similar words for `sendmsg`/`recvmsg`.
> The data transfers performed by readv() and writev() are atomic: [ .. ]